src/EncompassRest/DirtyDictionary.cs

```diff
@@ -0,0 +1,122 @@
+
+        public int Count => _dictionary.Count;
+
+        public bool IsReadOnly => false;
+
+        public bool Dirty
+        {
+            get
+            {
+                return _dictionary.Any(pair => pair.Value.Dirty);
+            }
+            set
+            {
+                foreach (var pair in _dictionary.ToList())
+                {
+                    var pairValue = pair.Value;
+                    pairValue.Dirty = value;
+                    _dictionary[pair.Key] = pairValue;
+                }
+            }
+        }
+
+        public DirtyDictionary()
+        {
+        }
+
+        public DirtyDictionary(IDictionary<TKey, TValue> dictionary)
+        {
+            if (dictionary != null)
+            {
+                foreach (var pair in dictionary)
+                {
+                    Add(pair);
+                }
+            }
+        }
+
+        public void Add(TKey key, TValue value) => _dictionary.Add(key, value);
+
+        public void Add(KeyValuePair<TKey, TValue> item) => Add(item.Key, item.Value);
+
+        public void Clear() => _dictionary.Clear();
+
+        public bool Contains(KeyValuePair<TKey, TValue> item)
+        {
+            DirtyValue<TValue> value;
+            return _dictionary.TryGetValue(item.Key, out value) && EqualityComparer<TValue>.Default.Equals(item.Value, value);
+        }
+
+        public bool ContainsKey(TKey key) => _dictionary.ContainsKey(key);
+
+        public void CopyTo(KeyValuePair<TKey, TValue>[] array, int arrayIndex)
+        {
+            Preconditions.NotNull(array, nameof(array));
+            Preconditions.LessThan(arrayIndex, nameof(arrayIndex), array.Length, $"{nameof(array)}.Length");
+            Preconditions.GreaterThanOrEquals(array.Length - arrayIndex, $"{nameof(array)}.Length - {nameof(arrayIndex)}", Count, nameof(Count));
+
+            var i = 0;
+            foreach (var pair in _dictionary)
+            {
+                array[arrayIndex + i] = new KeyValuePair<TKey, TValue>(pair.Key, pair.Value);
+                ++i;
+            }
+        }
+
+        public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
+        {
+            foreach (var pair in _dictionary)
+            {
+                yield return new KeyValuePair<TKey, TValue>(pair.Key, pair.Value);
+            }
+        }
+
+        public bool Remove(TKey key) => _dictionary.Remove(key);
+
+        public bool Remove(KeyValuePair<TKey, TValue> item) => Contains(item) && Remove(item.Key);
+
+        public bool TryGetValue(TKey key, out TValue value)
+        {
+            DirtyValue<TValue> dirtyValue;
+            var success = _dictionary.TryGetValue(key, out dirtyValue);
+            value = success ? (TValue)dirtyValue : default(TValue);
+            return success;
+        }
+
+        IEnumerator IEnumerable.GetEnumerator() => ((IDictionary<TKey, TValue>)_dictionary).GetEnumerator();
```

Comment:
Need to fix

Reply:
what is broken?

Reply:
`_dictionary` does not implement `IDictionary<TKey, TValue>`. It implements `IDictionary<TKey, DirtyValue<TValue>>`. I just need to change this to call `GetEnumerator`.